libraries/Firebase/examples/FirebasePush_ESP8266/FirebasePush_ESP8266.ino

```diff
@@ -0,0 +1,45 @@
+//
+// Copyright 2015 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <ESP8266WiFi.h>
+#include <WiFiClientSecure.h>
+
+#include <Firebase.h>
+
+WiFiClientSecure wifiClient;
+
+int pushButtonPin = 1;
+int previousState = 1;
+
+void setup() {
+    pinMode(pushButtonPin, INPUT);
+    digitalWrite(pushButtonPin, HIGH);
+ 
+    Serial.begin(9600);
+
+    WiFi.begin("my-ssid", "my-password");
+    Firebase.begin(wifiClient, "my-firebase.firebaseio.com");
```

Comment:
after chatting with @gguuss, instead of passing a `Client *` we could have Firebase return the HTTP requests as a string, and let the user send the request and receive the response.